packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -73,7 +73,7 @@ abstract class Route<T> {
   /// See also:
   ///
   /// * [Form], which provides an `onWillPop` callback that uses this mechanism.
-  Future<bool> willPop() async => true;
+  Future<bool> willPop() async => !isFirst;
```

Comment:
Although the docs didn't explain this before, it might be a good idea to explain the method's default behavior.